app/exerlog/lib/src/widgets/gradient_button.dart

```diff
@@ -0,0 +1,65 @@
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+
+class RaisedGradientButton extends StatefulWidget {
+  RaisedGradientButton({
+    required this.onPressed,
+    required this.child,
+    this.width = double.infinity,
+    this.height = 50.0,
+    this.radius = 30,
+    this.gradient,
+  });
+
+  final FutureOr Function() onPressed;
+  final Widget child;
+  final double width;
+  final double height;
+  final double radius;
+  final Gradient? gradient;
+
+  @override
+  _RaisedGradientButtonState createState() => _RaisedGradientButtonState();
+}
+
+class _RaisedGradientButtonState extends State<RaisedGradientButton> {
+  bool _isLoading = false;
+
+  @override
+  Widget build(BuildContext context) {
+    return SizedBox(
+      height: widget.height,
+      width: widget.width,
+      child: DecoratedBox(
+        decoration: BoxDecoration(
+          gradient: widget.gradient ??
+              LinearGradient(
+                colors: <Color>[
+                  Color(0xFF34D1C2),
+                  Color(0xFF31A6DC),
+                ],
+              ),
+          borderRadius: BorderRadius.circular(widget.radius),
+        ),
+        child: TextButton(
+          onPressed: _isLoading

```

Comment:
Maybe move this method to the bottom of the widget, and then just call it here in the onPressed. This way the ui and logic is separated a little more and its easier to get an overview of all the methods a widget can execute.